substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -4020,6 +4055,39 @@ impl<T: Config> Pallet<T> {
 		T::StakeAdapter::total_balance(Pool::from(Self::generate_bonded_account(pool_id)))
 			.unwrap_or_default()
 	}
+
+	fn do_reap_member(
+		member_account: T::AccountId,
+		pool_member: &PoolMember<T>,
+		bonded_pool: &BondedPool<T>,
+		num_slashing_spans: u32,
+	) {
+		ClaimPermissions::<T>::remove(&member_account);
+		PoolMembers::<T>::remove(&member_account);
+
+		let member = Member::from(member_account.clone());
+		let pool = Pool::from(bonded_pool.bonded_account());
+
+		// Ensure any dangling delegation is withdrawn.
+		let dangling_withdrawal = match T::StakeAdapter::member_delegation_balance(member.clone()) {
```

Comment:
The general approach looks fine to me. Good next step would be to write an integration test ([delegate-stake](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/nomination-pools/test-delegate-stake/src/lib.rs) and [transfer stake](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/nomination-pools/test-transfer-stake/src/lib.rs)) to ensure the stake is updated and storages are cleaned as expected.